packages/sdk/src/solana/programs/nft-collection.ts

```diff
@@ -427,35 +430,131 @@ export class NFTCollection {
   /**
    * Update the creators of the collection
    * @param creators - the creators to update
+   * @param updateAll - whether or not to retroactively update the creators of all past NFTs
    */
-  async updateCreators(creators: CreatorInput[]) {
-    const tx = await this.metaplex
-      .nfts()
-      .update({
-        nftOrSft: await this.getCollection(),
-        creators: enforceCreator(creators, this.metaplex.identity().publicKey),
-      })
-      .run();
-    return {
-      signature: tx.response.signature,
-    };
+  async updateCreators(
+    creators: CreatorInput[],
+    updateAll: boolean = false,
+  ): Promise<TransactionResult[]> {
+    if (updateAll) {
+      const txs: TransactionBuilder[] = [];
+
+      txs.push(
+        this.metaplex
+          .nfts()
+          .builders()
+          .update({
+            nftOrSft: await this.getCollection(),
+            creators: enforceCreator(
+              creators,
+              this.metaplex.identity().publicKey,
+            ),
+          }),
+      );
+
+      // TODO: Don't use a get all here, all we need is mint addresses :(
```

Comment:
we should break getAll() into 2 functions tbh, but yeah part of the optimization work

Reply:
Doing this in my other branch w/ totalCount